### PR TITLE
[DPE-2319] Update CI/CD to follow new discourse-gatekeeper logic

### PR DESCRIPTION
**Pull request by taurus-forever**
## Issue

The sync discourse -> git didn't work in full. Fixing it here based on https://github.com/canonical/kafka-operator/pull/133/files

## Solution

The recent changes to discourse-gatekeeper requires new CI/CD logic.

### CODE REVIEW

File: .github/workflows/sync_docs.yaml
Comment (taurus-forever) [inline .github/workflows/sync_docs.yaml:17]: Should/will be replaced with https://github.com/canonical/discourse-gatekeeper once published there.

**Review by taurus-forever — COMMENTED** [review]
Will be copied to all SQL repositories after the successful testing here.
TODO AFTER merging this PR:
 * delete git tag `upload-charm-docs/base-content`
 * abandon https://github.com/canonical/mysql-k8s-operator/pull/268

File: .github/workflows/ci.yaml
Comment (taurus-forever) [reply .github/workflows/ci.yaml:129]: it should be here (c) @deusebio 
Enrico, maybe it worth to wrote it as explanation somewhere? In `data-platform-workflows`?

File: .github/workflows/sync_docs.yaml
Comment (taurus-forever) [reply .github/workflows/sync_docs.yaml:9]: @deusebio please consider to apply on https://github.com/canonical/kafka-operator/pull/133/files (I am happy to copy it here).

**Comment by deusebio** [comment]
It took me a bit longer for responding as I have been discussing the exception that is thrown on the action with the team in IS. The exception is because the tool does not support at the moment that group rows (a line in the navigation table that has child items in the hierarchy below) has content (referencing to a dicourse topic). Weirdly enough, this has never been requested or needed in other documentations. 

**TLDR** my suggestion would be to change the navigation table as below. 

Basically, the following would work:

```
| 2 | h-upgrade | [Upgrade]() |
| 3 | h-upgrade-introduction | [Introduction](/t/upgrade-intro) |
| 3 | h-major-upgrade | [Major upgrade](/t/11750) |
| 3 | h-major-rollback | [Major rollback](/t/11751) |
``` 

but the following (which his what you have here) would not

```
| 2 | h-upgrade | [Upgrade](/t/upgrade-intro) |
| 3 | h-major-upgrade | [Major upgrade](/t/11750) |
| 3 | h-major-rollback | [Major rollback](/t/11751) |
```

The reason is that it isn't clear how to implement a level change in a file system if it isn't associated with a group (as navigation table has a 1:1 bindings with the naming/structure of the docs folder in the file system). We have discussed some ways (e.g. having a `index.md` file holding the content in a directory named with the group naming), but this needs moderate effort to be implemented, and won't land straight away. 

Hence, my suggestion. :D

File: .github/workflows/ci.yaml
Comment (deusebio) [reply .github/workflows/ci.yaml:129]: @taurus-forever , yes, I believe it would be sensible to add that to the documentation of the project

File: .github/workflows/sync_docs.yaml
Comment (deusebio) [reply .github/workflows/sync_docs.yaml:9]: Up to you whether it is worth failing the release workflow because of documentation misalignment. I'm fine with either

Comment (paulomach) [reply .github/workflows/sync_docs.yaml:9]: +1 to `s/workflow_call/push-to-main`

File: .github/workflows/sync_docs.yaml
Comment (paulomach) [reply .github/workflows/sync_docs.yaml:17]: What's keeping it from moving to canonical org?

**Comment by taurus-forever** [comment]
> ... Hence, my suggestion. :D

Applied as requested:
https://charmhub.io/mysql/docs/h-upgrade-intro
https://charmhub.io/mysql-k8s/docs/h-upgrade-intro

File: .github/workflows/sync_docs.yaml
Comment (deusebio) [reply .github/workflows/sync_docs.yaml:17]: [This](https://github.com/deusebio/discourse-gatekeeper/commit/4f7532da97a88dca5dc244e0dc1d1117a23c37da) patch. Ideally, this is not needed if:
1. Two-ways sync is enabled (which however we can't scale more broadly until we got upper leadership ok)
2. [This] spec is finally implemented (which would structure sync between Navigation Table and contents Index robust and reproducible

Maintainer of Discourse Gatekeeper did not want to merge the patch, since it will hopefully be solved once (2) is finalized. But for the time being, if one-way sync wants to be applied to our repo, this is the way to go I suppose. :(

File: .github/workflows/ci.yaml
Comment (deusebio) [reply .github/workflows/ci.yaml:129]: BTW, I have raised a PR on the Discourse Gatekeeper repository about this (but also improving the documenation which was somewhat out-of-sync with latest decisions)

https://github.com/canonical/upload-charm-docs/pull/213

**Review by deusebio — APPROVED** [review]
LGTM

**Review by paulomach — APPROVED** [review]
good to go

**Comment by paulomach** [comment]
Changes moved to https://github.com/canonical/mysql-k8s-operator/pull/336

**Pull request closed without merging**
